src/logion/model/locrequest.model.ts

```diff
@@ -807,21 +856,20 @@ export class LocRequestFactory {
         request.locType = description.locType;
         request.createdOn = description.createdOn;
         const userIdentity = description.userIdentity;
-        if (userIdentity !== undefined) {
-            request.userIdentity = new EmbeddableUserIdentity();
-            request.userIdentity.firstName = userIdentity.firstName;
-            request.userIdentity.lastName = userIdentity.lastName;
-            request.userIdentity.email = userIdentity.email;
-            request.userIdentity.phoneNumber = userIdentity.phoneNumber;
-        }
-        const userPostalAddress = description.userPostalAddress
-        if (userPostalAddress !== undefined) {
-            request.userPostalAddress = new EmbeddablePostalAddress();
-            request.userPostalAddress.line1 = userPostalAddress.line1;
-            request.userPostalAddress.line2 = userPostalAddress.line2;
-            request.userPostalAddress.postalCode = userPostalAddress.postalCode;
-            request.userPostalAddress.city = userPostalAddress.city;
-            request.userPostalAddress.country = userPostalAddress.country;
+        const userPostalAddress = description.userPostalAddress || {
+            line1: "",
+            line2: "",
+            postalCode: "",
+            city: "",
+            country: ""
+        }
+        if (request.locType === 'Identity' && userIdentity !== undefined) {
```

Comment:
`userIdentity` *must* be defined in this case, no?

Reply:
Yes, correct. I adapt the code to better reflect that.